### PR TITLE
Fixes around makefile

### PR DESCRIPTION
**Pull request by mmattel**
This PR adds some fixes and improvements around makefile:

- back-renaming file `validator.js` to `xref-validator.js` to be inline with the antora original
- using that name in the makefile
- adding help text for `check-prose`
- adding `write-good` (check-prose) to the dependencies
(discussed and agreed with @settermjd)
- adding a dash in front of the antora commands so that the echo afterwards will always be printed
(optical reasons only)
- adding `--stacktrace` (show issue details in case) to the antora command
- adding `make html` to build-the-docs and `make validate` to broken-link-checker

### CODE REVIEW

**Comment by tboerger** [comment]
Clean got to be removed, this can only be done by the clean task, otherwise the pdf documents are removed.

**Comment by mmattel** [comment]
@tboerger, removed as suggested

File: docs/checking-broken-links.md [hunk 1]
```diff
@@ -24,6 +24,18 @@ It is automatically installed when you run ``yarn install`` for the Antora setup
 
 To use it, you need to pass the custom generator (in `./generators/xref-validator.js`) to the `generate` command, as in the example below.
 
+#### Using Make
+
+This is the easiest way validating the documentation using predefined settings.
```
Comment (settermjd) [inline docs/checking-broken-links.md]: This should be: "This is the easiest way to validate the documentation, using predefined settings."

File: docs/checking-broken-links.md [hunk 2]
```diff
@@ -24,6 +24,18 @@ It is automatically installed when you run ``yarn install`` for the Antora setup
...
+
+```
+make validate
+```
+
+#### Using the Antora Tools On The Command-Line
+
+If you want to use own settings, you can run the command with it’s necessary parameters manually.
```
Comment (settermjd) [inline docs/checking-broken-links.md]: See the recommendation above.

File: Makefile [hunk 3]
```diff
@@ -48,7 +49,7 @@ clean:
 .PHONY: validate
 validate:
 	@echo "Validating xref links of all manuals defined within configuration"
-	antora generate --pull --cache-dir $(CACHE_DIR) --redirect-facility $(REDIRECTS) --generator ./generators/validator $(PLAYBOOK)
+	-antora generate --pull --cache-dir $(CACHE_DIR) --redirect-facility $(REDIRECTS) --stacktrace --generator=./generators/xref-validator.js $(PLAYBOOK)
```
[the remark this answers is not in the document]
Comment (mmattel) [reply Makefile]: the dash will be removed when the command runs but it tells make to continue (echo in that case) when antora returns with non zero.

[the remark this answers is not in the document]
Comment (settermjd) [reply Makefile]: Is that necessary, as there's only one command?

[the remark this answers is not in the document]
Comment (mmattel) [reply Makefile]: Yes, if there is an non zero exit of the antora command, make does not continue.
Using the dash, tells make to continue and in that particular case it just echoes a new line which his an optical support for better readability.

**Comment by mmattel** [comment]
@settermjd incorporated all your change requests squashed and pushed.

**Comment by tboerger** [comment]
Since the branch anyway got to be updated, maybe you can change REDIRECTS to static instead of nginx?

**Comment by mmattel** [comment]
@tboerger I added the change as requested and a note regarding https://github.com/owncloud/docs/issues/444

**Pull request merged**